rust/nasl-interpreter/src/scanner/running_scan.rs

```diff
@@ -0,0 +1,297 @@
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, RwLock,
```

Comment:
shouldn't we use tokio RwLock when we run in async contexts?

Reply:
oh - good catch. we should!